sys/riscv/riscv/pmap.c

```diff
@@ -4611,19 +4611,17 @@ pmap_clear_modify(vm_page_t m)
 	int md_gen, pvh_gen;
 
 	KASSERT((m->oflags & VPO_UNMANAGED) == 0,
-	    ("pmap_clear_modify: page %p is not managed", m));
+	    ("%s: page %p is not managed", __func__, m));
 	vm_page_assert_busied(m);
 
-	if (!pmap_page_is_write_mapped(m))
-	        return;
-
 	/*
 	 * If the page is not PGA_WRITEABLE, then no PTEs can have PG_M set.
 	 * If the object containing the page is locked and the page is not
 	 * exclusive busied, then PGA_WRITEABLE cannot be concurrently set.
 	 */
```

Comment:
```suggestion

```
This comment should be removed, it is no longer accurate and not present in other pmap implementations.

In fact it seems this comment and check were missed in the previous change, see: 638f867814.

Reply:
Should we do this here, or as a separate commit?

Reply:
The user.name and user.email are correct.